server-spi/src/main/java/org/keycloak/provider/ProviderConfigProperty.java

```diff
@@ -162,13 +202,43 @@ public void setDefaultValue(Object defaultValue) {
      * @return
      */
     public List<String> getOptions() {
-        return options;
+        return Optional.ofNullable(options).orElse(List.of());
+    }
+
+    /**
+     * Resolves the options based on the given {@code session}. If no option resolver was set via {@link #setOptionsResolver(Function)}
+     * any static option set via {@link #setOptions(List)} is returned.
+     *
+     * @param session the current (not null) session.
+     * @return the options for this property or an empty list if no options is set
+     * @throws IllegalArgumentException if the {@code session} is null
+     */
+    public List<String> getOptions(KeycloakSession session) {
```

Comment:
For the sake of simplicity, we are opinionated about only specific options (`type` and `options`) to allow resolving their values from a session.

A more flexible approach is to change all the types (more than 200 classes) where the `ConfiguredProvider` is implemented to allow obtaining all the provider properties based on a session (e.g.: changing the signature of the `getConfiguredProperties` method). Not sure if want to do that now but as a follow-up.